src/main/java/graphql/schema/idl/SchemaPrinter.java

```diff
@@ -792,6 +804,25 @@ private void printAsAst(PrintWriter out, TypeDefinition<?> definition, List<? ex
         out.print('\n');
     }
 
+    /**
+     * This will print out a runtime graphql schema block using its AST definition.  This
+     * must be guarded by a called to {@link #shouldPrintAsAst(SchemaDefinition)}
+     *
+     * @param out        the output writer
+     * @param definition the AST schema definition
+     * @param extensions a list of schema definition extensions
+     */
+    private void printAsAst(PrintWriter out, SchemaDefinition definition, List<SchemaExtensionDefinition> extensions) {
+        out.printf("%s\n", AstPrinter.printAst(definition));
+        if (extensions != null) {
+            for (SchemaExtensionDefinition extension : extensions) {
+                out.printf("\n%s\n", AstPrinter.printAst(extension));
+            }
+        }
+        out.print('\n');
+    }
+
```

Comment:
Similar to type definition version above. This prints everything including directives and description.